### DISCUSSION TITLE
Is it possible to make cell data async

### DISCUSSION THREAD

**Discussion by mikiyasET** [KevinVandy/material-react-table#298] [General]
Here is what I was trying to do

```
const columns = useMemo(
      () => [
      {
          header: 'Name',
          accessorKey: 'name',
      },
      {
          header: 'Description',
          accessorKey: 'description',
      },
      {
          header: 'image',
          accessorKey: 'image',
          Cell: async ({cell}: any) => <img
              src={await Service.Run().VendorsCategoryService.getVendorsCategoryImage(cell.getValue())} alt="Image"/>,
      }
  ],[]);
```

The error occurred after I add the following code

`Cell: async ({cell}: any) => <img src={await Service.Run().VendorsCategoryService.getVendorsCategoryImage(cell.getValue())} alt="Image"/>,`

I'm facing error

<img width="1451" alt="image" src="https://user-images.githubusercontent.com/25723129/210929280-934ee524-7394-4d75-871b-8b55d9b8f4a6.png">

I'm trying to load image to the table and to get the image I need have a token that's why I'm not specifying the url directly. can someone help...

**Comment by KevinVandy** [comment]
Not async like that, but just do a normal cell render with no async function with and access data from a useState hook or something. Add that state to the columns useMemo dependency array.

**Reply by mikiyasET** [reply]
It would be great if it was handled in the cell don't you think?

**Reply by KevinVandy** [reply]
As far as I understand, the syntax you are proposing is pretty much impossible in react. You can't have async components like that. That is until Next.js allows it in the future for server components, but that kind of think isn't possible here

**Reply by mikiyasET** [reply]
> As far as I understand, the syntax you are proposing is pretty much impossible in react. You can't have async components like that. That is until Next.js allows it in the future for server components, but that kind of think isn't possible here

thanks.
